### ISSUE TITLE
OnSongChanged() example hangs on changing to SetList view when setlist is empty

### ISSUE THREAD

**Issue by daveboulden** [gigperformer/gp-extension-cpp#2]
I am using this example as the basis for a new extension. During testing I was getting a hang when changing to SetList mode. this was because I had and empty Gig file and there were no songs. I change the `OnSongChanged()` code to:

```
void LibMain::OnSongChanged(int oldIndex, int newIndex)
{
    if (getSongCount() > 0)
    {
        Ignore(oldIndex);
        consoleLog("Song changed from C++ example");
        std::string name = getSongName(newIndex);
        consoleLog("New song is called " + name);
    }
}
```

**Comment by sleiner** [comment]
Good catch, thanks so much! :-)

Two questions:
 - Do you want to open a pull request with this patch?
 - Out of interest: Did you catch which values `oldIndex` and `newIndex` have assigned?

**Comment by daveboulden** [comment]
I'm afraid my skills with Git don't quite stretch to pull requests, I only ever try and do the basics with Git.

Sorry, but I didn't catch what the values were. In the end I removed the `OnSongChanged()` override and its associate callback registration as it wasn't needed for my extension.

**Comment by sleiner** [comment]
@daveboulden First of all: Sorry for taking so long! I have just tried to reproduce the issue (on Gig Performer 4.5.8, the latest release) - and couldn't get it to hang...

  - I could not reproduce receiving an `OnSongChanged` callback if there were no songs.
  - When I create the first song in a gig, I _do_ get `OnSongChanged(oldIndex = -1, newIndex = 0)` which seems fine to me.
  - `getSongName()` works as expected for `newIndex` >= 0.
  - `getSongName(-1)` (which would never be called from the callback implementation as it stands) returns an empty string; still no hanging.

Can you reproduce the problem with the extension template (as opposed to your own extension)? And if so, which build settings, Gig Performer version, and OS version did you use?

**Comment by daveboulden** [comment]
@sleiner I think the hang may have been related to something else entirely. At the same time, I was testing OSC integration and had a bad configuration (same in & out port number) and that caused a hang and is a known scenario in GP. 
This was some time after I made this change and looking back retrospectively, I believe I attributed this change to being the remedy, when it was actually changing the OSC config.
